[AY1920S1-CS2113-W17-G1]COMPal

sholihinK opened the pull request:
@jaedonkey
@yueyeah
@LTPZ
@Catherinetan99

seanieyap commented on src/main/java/compal/commons/Compal.java:
```diff
@@ -0,0 +1,157 @@
+
+    //***CONSTRUCTORS***------------------------------------------------------------------------------------------------
+    //------------------------------------------------------------------------------------------------------------------
+    //----------------------------------------------------------------------------------------------------------------->
```
Avoid doing this. Try to strictly follow the coding standards for commenting.
https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#comments

seanieyap commented on src/main/java/compal/commons/Compal.java:
```diff
@@ -0,0 +1,157 @@
+    //------------------------------------------------------------------------------------------------------------------
+    //----------------------------------------------------------------------------------------------------------------->
+
+    //----------------------->
```
As mentioned above, this definitely waste too many lines and makes code unnecessarily long and harder to read.

seanieyap commented on src/main/java/compal/commons/Compal.java:
```diff
@@ -0,0 +1,157 @@
+        if (reminder.isEmpty()) {
+            ui.printg("You currently have no tasks that have reminders set or are due within "
+                            + numberOfDays + " days!",
+                    "verdana", 15, Color.DARKGREEN);
```
Try to apply more SLAP such that you would not need comments to tell the reader what you are trying to do. The comments, as stated on the module website, is unnecessary.

seanieyap commented on src/main/java/compal/commons/Compal.java:
```diff
@@ -0,0 +1,157 @@
+            int counter = 1;
+            for (Task t : reminder) {
+                if (t.getPriority().equals(Task.Priority.high)) {
+                    //compal.ui.printg(counter + ". " + t.toString(),"verdana",15, Color.RED);
```
Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

seanieyap commented on src/main/java/compal/commons/Messages.java:
```diff
@@ -0,0 +1,59 @@
+/**
+ * Container for user visible messages.
+ */
+
```
Extra line can be deleted.

seanieyap commented on src/main/java/compal/commons/Messages.java:
```diff
@@ -0,0 +1,59 @@
+
+public class Messages {
+
+
```
Extra lines can be deleted.

seanieyap commented on src/main/java/compal/logic/commands/AcadCommand.java:
```diff
@@ -0,0 +1,214 @@
+
+    /**
+     * Constructs RecurTaskCommand object.
+     *
```
Extra line can be deleted.

seanieyap commented on src/main/java/compal/logic/commands/DoneCommand.java:
```diff
@@ -0,0 +1,59 @@
+            compal.ui.printg(MESSAGE_MISSING_COMMAND_ARG);
+            throw new Compal.DukeException(MESSAGE_MISSING_COMMAND_ARG);
+        }
+        //Compal.tasklist.taskDone(userIn);
```
Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

seanieyap commented on src/main/java/compal/logic/commands/EditCommand.java:
```diff
@@ -0,0 +1,75 @@
+
+            default:
+                System.out.println("EditCommand:LOG: Default case");
+            }
```
Indentation problem here? Perhaps an two "ifs" would better to be used here instead of two switch; would perhaps be easier to read.

seanieyap commented on src/main/java/compal/logic/commands/EditCommand.java:
```diff
@@ -0,0 +1,75 @@
+            default:
+                System.out.println("EditCommand:LOG: Default case");
+            }
+
```
Use blank lines sparingly. It's good to have blank lines to separate logical code blocks, but too many blank lines will increase the height of the code, which increases the amount of scrolling.

seanieyap commented on src/main/java/compal/logic/commands/EditCommand.java:
```diff
@@ -0,0 +1,75 @@
+
+        }
+
+        //save the changes
```
Unnecessary comment, try to comment sparingly

seanieyap commented on src/main/java/compal/logic/commands/FindCommand.java:
```diff
@@ -0,0 +1,64 @@
+                compal.ui.printg(task.toString());
+                isEmpty = false;
+            }
+        }
```
Code logic is correct but it can be made more efficient.

seanieyap commented on src/main/java/compal/logic/commands/FindFreeSlotCommand.java:
```diff
@@ -0,0 +1,138 @@
+    }
+
+    /**
+     * Cat to add javadoc desc.
```
Comment descriptions should be understandable.

seanieyap commented on src/main/java/compal/logic/commands/RecurTaskCommand.java:
```diff
@@ -0,0 +1,208 @@
+     *
+     * @param d Compal.
+     */
+    public RecurTaskCommand(Compal d) {
```
"d" can be named "compal" to give more context for other developers.

seanieyap commented on src/main/java/compal/logic/parser/ParserManager.java:
```diff
@@ -0,0 +1,202 @@
+                    compal.ui.printg(MESSAGE_INVALID_COMMAND);
+                    throw new Compal.DukeException(MESSAGE_INVALID_COMMAND);
+                }
+            }
```
Indentation issue on switch statements. Perhaps an if else here would be easier to read.

seanieyap commented on src/main/java/compal/logic/parser/ParserManager.java:
```diff
@@ -0,0 +1,202 @@
+     */
+    public void setStatus(String status) {
+        this.status = status;
+        stage = 0; //reset stage everytime status is changed
```
comment is unnecessary

seanieyap commented on src/main/java/compal/model/tasks/Task.java:
```diff
@@ -0,0 +1,395 @@
+     * @return Status icon (tick or cross) of task.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718");
```
"Magic strings" should be avoided. Especially so when you have put in so much effort to avoid them but make the mistake ehre.

seanieyap commented on src/main/java/compal/model/tasks/Task.java:
```diff
@@ -0,0 +1,395 @@
+        return stringDate;
+    }
+
+    /**
+     * Gets durationHour of task.
+     *
+     * @return Hour duration of task.
+     */
+
```
??

seanieyap commented on src/main/java/compal/model/tasks/Task.java:
```diff
@@ -0,0 +1,395 @@
+     */
+    public void calculateAndSetPriorityScore() {
+        long score;
+        switch (priority) {
+
+        case high:
+            score = 100;
+            break;
+        case medium:
+            score = 50;
+            break;
+        case low:
+            score = 20;
+            break;
+        default:
+            score = 0;
+
+        }
```
Indentation issue.

seanieyap commented on src/main/java/compal/model/tasks/Task.java:
```diff
@@ -0,0 +1,395 @@
+        Date d = new Date();
+        long diff = d.getTime() - this.date.getTime();
+        long diffHours = diff / (60 * 60 * 1000);
+        //System.out.println("Task:LOG: Difference is " + diffHours);
```
Debugging code should be removed entirely

seanieyap commented on src/main/java/compal/model/tasks/TaskList.java:
```diff
@@ -0,0 +1,221 @@
+        idBitSet.clear(id);
+    }
+
+
```
Extra lines

seanieyap commented on src/main/java/compal/model/tasks/TaskList.java:
```diff
@@ -0,0 +1,221 @@
+    }
+
+
+
```
Formatting issue, extralines and etc. Also, avoid printing stacktrace.

seanieyap commented on src/test/java/compal/logic/commands/AcadCommandTest.java:
```diff
@@ -0,0 +1,79 @@
+        String newDate = "21/10/2019";
+        SimpleDateFormat format = new SimpleDateFormat("dd/MM/yyyy");
+        Date d = null;
+        Date newD = null;
```
Try to use intuitive variable names. "d" is definitely not acceptable.

seanieyap commented on src/test/java/compal/logic/commands/ByeCommandTest.java:
```diff
@@ -0,0 +1,206 @@
+            String time = "/priority";
+            byeCommand.getPriority(time);
+        } catch (NullPointerException e) {
+            //ignore the null pointer exception because ui is not welly declared in this test
```
Appropriate.

seanieyap reviewed (commented):
Overall good work but it is obvious that not everying adhere to the same coding style. Also, there is a huge habit of leaving commented dead and unused code undeleted. However, the software engineering principles seem to be followed and tests seem to be well underway since the skeleton are already in place.

seanieyap commented:
Please code this PR after reading my comments.

Catherinetan99 committed:
Merge pull request #208 from Catherinetan99/legit

Added Junit Testing

Catherinetan99 committed:
Fixed some bugs

Catherinetan99 committed:
Fixed codacy issues

Catherinetan99 committed:
Merge pull request #209 from Catherinetan99/legit

Fixed some bugs

LTPZ committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	src/main/java/compal/logic/command/ViewCommand.java
#	src/main/java/compal/logic/command/ViewReminderCommand.java
#	src/main/java/compal/logic/parser/CommandParser.java

LTPZ committed:
update author&log

LTPZ committed:
Merge remote-tracking branch 'upstream/master'

LTPZ committed:
update junit & help & ug

LTPZ committed:
pass checkstyle
add help for delete function

LTPZ committed:
fix Codacy

LTPZ committed:
fix Codacy again

yueyeah committed:
Merge pull request #214 from LTPZ/master

Junit & log & UG update

LTPZ committed:
add delete in general help

sholihinK committed:
Update AboutUs.md

LTPZ committed:
Help bug fixed

LTPZ committed:
Merge pull request #215 from LTPZ/master

Help improved

sholihinK committed:
Update README.md

sholihinK committed:
Uploaded COMPal DG.pdf to test coverall

sholihinK committed:
Merge pull request #216 from SholihinK/test

Added dg.

sholihinK committed:
Update README.md

sholihinK committed:
Delete COMPal Developer Guide(V1.3).pdf

Catherinetan99 committed:
Added more Junit testing

sholihinK committed:
Merge pull request #218 from Catherinetan99/legit

Added more Junit testing

sholihinK committed:
Uploaded DG.pdf

Updated as of 30/10/2019

sholihinK committed:
Update README.md

sholihinK committed:
Update gradle to have opencsv.

sholihinK committed:
Basic export command setup

sholihinK committed:
Add files via upload

sholihinK committed:
Update README.md

sholihinK committed:
updated output for export command

sholihinK committed:
Added import Command

sholihinK committed:
Update import and export command

sholihinK committed:
update gitignore

sholihinK committed:
added export test

sholihinK committed:
making codacy happpy

sholihinK committed:
making travis happy

sholihinK committed:
making codacy happier

sholihinK committed:
Merge pull request #219 from SholihinK/importExport

Added Import export

Catherinetan99 committed:
Fixed bugs

Catherinetan99 committed:
Fixed checkstyle issues

Catherinetan99 committed:
Fixed codacy

Catherinetan99 committed:
Fixed codacy

jaedonkey committed:
Merge pull request #34 from AY1920S1-CS2113T-W17-1/master

update

Catherinetan99 committed:
Updated JUnit tests

Catherinetan99 committed:
Fixed codacy

Catherinetan99 committed:
Commented out parset JUnit test for now

Catherinetan99 committed:
Merge pull request #220 from Catherinetan99/legit

Fixed bugs

sholihinK committed:
Updated View test

sholihinK committed:
Merge branch 'importExport'

sholihinK committed:
making codacy happy

sholihinK committed:
Merge pull request #221 from SholihinK/ImportExport2

Updated view command

jaedonkey committed:
Merge pull request #35 from AY1920S1-CS2113T-W17-1/master

update

jaedonkey committed:
fixed edit bug

LTPZ committed:
Merge pull request #222 from jaedonkey/master

fixed edit bug

LTPZ committed:
fix help bug

yueyeah committed:
Update README.md

LTPZ committed:
Merge pull request #223 from LTPZ/master

fixed help bug

sholihinK committed:
added importCommandParserTest,fix display output.

sholihinK committed:
making travis happy

sholihinK committed:
making travis happy

sholihinK committed:
codacy bug fix

sholihinK committed:
Merge pull request #241 from SholihinK/ImportExport2

added importCommandParserTest,fix display output.

sholihinK committed:
update list output to be more specific.

sholihinK committed:
making travis happy

sholihinK committed:
Merge pull request #258 from SholihinK/ImportExport2

update list output to be more specific.

sholihinK committed:
Fix bug from PE dry Run.

sholihinK committed:
Apply suggestions from code review

sholihinK committed:
Merge pull request #259 from SholihinK/bugFix

Fix bug from PE dry Run.

sholihinK committed:
Added JUnit for import and export.

sholihinK committed:
Updated JUnit for export

sholihinK committed:
added help to export and import

sholihinK committed:
making travis ahppy

sholihinK committed:
makign travis happy

sholihinK committed:
Merge pull request #260 from SholihinK/Junit

Junit & help for export and import

sholihinK committed:
Test relative path for junit test

sholihinK committed:
Commented out test for file path

LTPZ committed:
Solve issue 253 226

LTPZ committed:
Merge pull request #261 from LTPZ/master

Solve issue 253 226

sholihinK committed:
Update gradle version

sholihinK committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-W17-1/main

sholihinK committed:
Merge branch 'master' into 1.4

sholihinK committed:
Added PPP

sholihinK committed:
Merge pull request #262 from SholihinK/1.4

Prep for v1.4 end

sholihinK committed:
Delete SholihinK.pdf

sholihinK committed:
Improve algo of adding deadline or events.

sholihinK committed:
Merge pull request #263 from SholihinK/imporveBigo

Improve algo of adding deadline or events.

yueyeah committed:
Merge pull request #52 from AY1920S1-CS2113T-W17-1/master

update yueyeah fork

LTPZ committed:
fix bugs

LTPZ committed:
update junit

LTPZ committed:
Merge pull request #265 from LTPZ/master

fix bugs

LTPZ committed:
solve issue #242

LTPZ committed:
Merge remote-tracking branch 'upstream/master'

LTPZ committed:
Merge pull request #266 from LTPZ/master

Fix bug

yueyeah committed:
Merge pull request #53 from AY1920S1-CS2113T-W17-1/master

Update yueyeah fork

sholihinK committed:
Update import and export to use deadline,event and set-reminder command instead of adding to tasklist directly.

sholihinK committed:
Updated even to handle duplicate events, updated import

sholihinK committed:
Update AboutUs.md

sholihinK committed:
Update README.md

sholihinK committed:
making codacy happy

sholihinK committed:
Merge branch 'updateImportExport' of https://github.com/SholihinK/main into updateImportExport

sholihinK committed:
travis and codacy issue fix

sholihinK committed:
test nPath

sholihinK committed:
Update test case

sholihinK committed:
Merge pull request #267 from SholihinK/updateImportExport

Prevent Duplicate event

sholihinK committed:
Update to handle extra param for command

sholihinK committed:
Merge pull request #270 from SholihinK/master

Update to handle extra param for command

jaedonkey committed:
Merge pull request #36 from AY1920S1-CS2113T-W17-1/master

update

jaedonkey committed:
added ppp

jaedonkey committed:
fixed edit bugs

jaedonkey committed:
Merge branch 'master' of https://github.com/jaedonkey/main

jaedonkey committed:
Merge pull request #37 from AY1920S1-CS2113T-W17-1/master

added ppp

jaedonkey committed:
fixed another edit bug

jaedonkey committed:
making codacy happier than sho does

jaedonkey committed:
making travis HAPPY pls be happy

jaedonkey committed:
making travis HAPPY pls be happy

jaedonkey committed:
Merge pull request #271 from jaedonkey/master

fixed edit bugs

LTPZ committed:
Fix extra para bug

LTPZ committed:
Make Codacy happy

LTPZ committed:
Merge pull request #272 from LTPZ/master

Fix extra parameter bug for deadline

yueyeah committed:
Merge pull request #54 from AY1920S1-CS2113T-W17-1/master

Update yueyeah fork

sholihinK committed:
Delete SholihinK.pdf

sholihinK committed:
Add files via upload

Finalized PPP submission for sholihinK

Catherinetan99 committed:
Added logging

Catherinetan99 committed:
Merge pull request #273 from Catherinetan99/legit

Added logging

sholihinK committed:
Update import command to give more error specific messages.

Catherinetan99 committed:
Update to handle extra param for command

Catherinetan99 committed:
Fixed codacy

sholihinK committed:
fix import bugs

sholihinK committed:
Fix testCase

yueyeah committed:
Made new diagrams for Logic Component, for DG

yueyeah committed:
Added guard against extra invalid parameters for EventCommandParser

yueyeah committed:
Added CompalUtilsTest

yueyeah committed:
Merge pull request #55 from AY1920S1-CS2113T-W17-1/master

Update yueyeah fork

yueyeah committed:
Merge pull request #276 from yueyeah/master

Added CompalUtilTest, allow detection of extra invalid parameter for Events

sholihinK committed:
Merge pull request #275 from SholihinK/importUp

Fix import not handling proper iCal

sholihinK committed:
Merge pull request #274 from Catherinetan99/legit

Update to handle extra param for command

jaedonkey committed:
Merge pull request #38 from AY1920S1-CS2113T-W17-1/master

update

jaedonkey committed:
added error throwing for underscores in desc

Catherinetan99 committed:
Added JUnit test for findfreeslot command parser

jaedonkey committed:
codacy you needy b*tch

jaedonkey committed:
Merge pull request #277 from jaedonkey/master

Underscore handling

jaedonkey committed:
Merge pull request #278 from Catherinetan99/legit

Added JUnit test for findfreeslot command parser

sholihinK committed:
Included to add data on creation

sholihinK committed:
Update test case

sholihinK committed:
Merge pull request #279 from SholihinK/withDate

Included to add data on creation

sholihinK committed:
Update SholihinK PPP

jaedonkey committed:
Merge pull request #39 from AY1920S1-CS2113T-W17-1/master

update

jaedonkey committed:
-fixed delete not working
-fixed edit invalid params
-added delete invalid params handling
-refined validKey method to check for duplicate keys/tokens

sholihinK committed:
Merge pull request #280 from jaedonkey/master

minor bug fixes

Catherinetan99 committed:
Added deadline and events to createData function

Catherinetan99 committed:
Fixed checkstyle issues

jaedonkey committed:
Merge pull request #281 from Catherinetan99/legit

Added deadline and events to createData function

jaedonkey committed:
Merge pull request #40 from AY1920S1-CS2113T-W17-1/master

update

sholihinK committed:
Update list to be more user freidnly

sholihinK committed:
codacy fix

yueyeah committed:
Merge pull request #56 from AY1920S1-CS2113T-W17-1/master

Update yueyeah fork

sholihinK committed:
Merge pull request #282 from SholihinK/updateList

Update list to be more user friendly

jaedonkey committed:
Merge pull request #41 from AY1920S1-CS2113T-W17-1/master

update

sholihinK committed:
Fix bug in list and bye taking in extra param or tokens

sholihinK committed:
Merge pull request #283 from SholihinK/updateList

Fix bug in list and bye taking in extra param or tokens

jaedonkey committed:
-fixed bug for slash char '/'

jaedonkey committed:
Merge branch 'master' of https://github.com/jaedonkey/main

jaedonkey committed:
Merge pull request #42 from AY1920S1-CS2113T-W17-1/master

update

jaedonkey committed:
-fixed wrong test input

jaedonkey committed:
Merge pull request #284 from jaedonkey/master

fixed slash bug

sholihinK committed:
Update for view daily task,parser and message.

sholihinK committed:
Update daily task tab to reflect the same output as view day in main tab

sholihinK committed:
fix travis

sholihinK committed:
Merge pull request #285 from SholihinK/updateList

Bug fix for view daily,parser and message.

LTPZ committed:
update help diagram

LTPZ committed:
Merge remote-tracking branch 'upstream/master'

yueyeah committed:
Merge pull request #57 from AY1920S1-CS2113T-W17-1/master

Update yueyeah fork

LTPZ committed:
update help

LTPZ committed:
Merge pull request #286 from LTPZ/master

update help

yueyeah committed:
Solved lack of invalid param bug for getIntInput

yueyeah committed:
Merge pull request #58 from AY1920S1-CS2113T-W17-1/master

Update yueyeah fork

yueyeah committed:
rectify codacy

yueyeah committed:
Merge branch 'master' of https://github.com/yueyeah/main

yueyeah committed:
Merge pull request #287 from yueyeah/master

Resolve lack of invalid param bug in getIntInput

sholihinK committed:
rename mainLauncer to mainLauncher

sholihinK committed:
Merge pull request #288 from SholihinK/master

rename mainLauncer to mainLauncher

sholihinK committed:
Remove unused imports + update sorting algo

sholihinK committed:
Merge pull request #289 from SholihinK/Push

Remove unused imports + update sorting algo

sholihinK committed:
Limit number of tasks to add at a time.

sholihinK committed:
Merge pull request #290 from SholihinK/Push

Limit number of tasks to add at a time.

sholihinK committed:
Delete SholihinK.pdf

sholihinK committed:
Add files via upload

sholihinK committed:
Add files via upload

sholihinK committed:
Delete [CS2113T-W17-1][COMPal]UserGuide.pdf

sholihinK committed:
Add files via upload

sholihinK committed:
Update README.md

sholihinK committed:
Delete [AY1920S1-CS2113T-W17-1]-COMPal-DG.pdf

sholihinK committed:
Delete DeveloperGuide.md

sholihinK committed:
Delete UserGuide.md

sholihinK committed:
Delete [AY1920S1-CS2113T-W17-1]-COMPal-UG.pdf

sholihinK committed:
Add files via upload

sholihinK committed:
Delete SholihinK.pdf

sholihinK committed:
Add files via upload

sholihinK committed:
Delete [CS2113T-W17-1][COMPal]DeveloperGuide.pdf

sholihinK committed:
Add files via upload

LTPZ committed:
My ppp

my ppp

LTPZ committed:
Update AboutUs.md

LTPZ committed:
Delete LTPZ.pdf

yueyeah committed:
yeah bois and gurls

LTPZ committed:
Add files via upload

yueyeah committed:
yeah bois and gurls

Catherinetan99 committed:
Added PPP

sholihinK committed:
Bug fix for testing bug found

sholihinK committed:
Update AboutUs.md

sholihinK committed:
Fix travis and testcase

Catherinetan99 committed:
Delete Catherinetan99.pdf

Catherinetan99 committed:
Add files via upload

sholihinK committed:
Merge pull request #291 from SholihinK/master

Fix bug

yueyeah committed:
yes

yueyeah committed:
yeah bois and gurls

yueyeah committed:
Merge pull request #59 from AY1920S1-CS2113T-W17-1/master

Update yueyeah fork

yueyeah committed:
Added functionality to parse NUSMODS link

yueyeah committed:
Basic Functionality of getting and parsing JSON from NUSMODS

yueyeah committed:
Accounted for Recess Week, fix checkstyle

yueyeah committed:
Allow storage into tasks.txt

yueyeah committed:
Added help for nusmods command

yueyeah committed:
Finished basic QuoteCommand

yueyeah committed:
Finished up QuoteCommand

yueyeah committed:
Added help section for quote

yueyeah committed:
Fixed test issues, by changing findfreeslot tests to year 2020

yueyeah committed:
Added more documentation and explanation for why cannot create exam
event

yueyeah committed:
Modify README.md

yueyeah committed:
Merge branch 'enhancement'

yueyeah committed:
Update README.md

yueyeah committed:
Optimised parsing of json timetable

yueyeah committed:
Rectified codacy

yueyeah committed:
Modified README.md

yueyeah committed:
Modified README.md

yueyeah committed:
Merge pull request #298 from yueyeah/master

Optimised NUSMODS parsing

yueyeah committed:
commented out the template data function

yueyeah committed:
removed certain quotes

yueyeah committed:
Merge pull request #300 from yueyeah/master

minor cleanup

yueyeah committed:
Experimentation with editing UI and Main

yueyeah committed:
Remove JavaFX versioning from fxml files to remove JavaFX version warnings

yueyeah committed:
Add documentation for MainLauncher

yueyeah committed:
Added comments explaining UiManager and fix checkstyle for
MainWindow

yueyeah committed:
Set theme jekyll-theme-architect

yueyeah committed:
Cleanup

yueyeah committed:
GitHub Pages test

yueyeah committed:
Merge branch 'master' of https://github.com/yueyeah/COMPal

yueyeah committed:
Fixed doc links

yueyeah committed:
Fix links 2. Will this finally work?

yueyeah committed:
Fixed typos and 1 last attempt at fixing image issue

yueyeah committed:
Merge pull request #60 from AY1920S1-CS2113T-W17-1/master

Pulling Merge Commits

yueyeah committed:
Merge pull request #301 from yueyeah/master

Website + UI Review
